app/models/email_address.rb

```diff
@@ -1,7 +1,52 @@
 class EmailAddress < ContactInfo
   include EmailAddressValidations
 
+  WHITELIST = [
+    # popular email providers globally
+    'gmail.com', 'outlook.com', 'yahoo.com', 'icloud.com', 'hotmail.com', 'aol.com',
+    # popular ones currently in our DB - as of Feb 2019
+    'rice.edu',
+    'openstax.org',
+    'uga.edu',
+    'vols.utk.edu',
+    'email.vccs.edu',
+    'ku.edu',
+    'bruinmail.slcc.edu',
+  ]
+
+  BLACKLIST = ['']
+
   email_validation_formats.each do |format|
     validates :value, format: format
   end
+
+  validate :mx_domain_validation
+
+  def domain
+    @domain ||= self.value.strip.split('@')[1] rescue ''
```

Comment:
Good question. That method should be `protected`.

Reply:
Rails uses Mail internally, so you could do: `Mail::Address.new(self.value).domain`

https://www.rubydoc.info/github/mikel/mail/Mail%2FAddress:domain

Reply:
Great, thanks @nathanstitt. Note, however, that it can still raise an exception when the email address is invalid (eg. from an actual test `bob@gma@il.com`) so I had to `rescue Mail::Field::IncompleteParseError`.